### PR TITLE
adding  Madgraph5_aMC@NLO cards for WWG NLO  background samples  and MCFM cards for ggZZto4E and ggZZto4mu background samples [Run 3]

### PR DESCRIPTION
**Pull request by sv3048**
Dear gen experts !

This is for the Run3 production of the WWG nlo process and ggZZto4e and ggZZto4mu background processes,

For WWG nlo only the CoM has been changed to 6800 from 6500 TeV in the run card. 
Regarding  ggZZto4e and ggZZto4mu processes along with COM, the PDF is also updated.

Please verify the cards and send your comments.

### CODE REVIEW

File: bin/MCFM/cards/MCFM+JHUGen/13p6TeV/MCFM_JHUGen_13TeV_ggZZtoELEL_BKG_NNPDF31.DAT [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+[Pdf selection]
+cteq6_l [pdlabel]
+4       [NGROUP, see PDFLIB]
+46      [NSET - see PDFLIB]
+'NNPDF31_nnlo_as_0118_mc_hessian_pdfas'			[LHAPDF group]
+0              [LHAPDF set]
+
+[Jet definition and event cuts]
+2.5d0		[m34min]
+13000d0		[m34max]
```
Comment (menglu21) [inline bin/MCFM/cards/MCFM+JHUGen/13p6TeV/MCFM_JHUGen_13TeV_ggZZtoELEL_BKG_NNPDF31.DAT:51]: could you please also change them to 13.6 tev, although it won't change much

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#*     run as ./bin/mg5_aMC  filename                       *
+#*                                                          *
+#************************************************************
+set loop_optimized_output True
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat:29]: the banner can be removed

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import model loop_sm
+define l+ = e+ mu+
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat:34]: these four lines can be removed

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+# Reweight flags to get scale dependence and PDF uncertainty           *
+# For scale dependence: factor rw_scale_up/down around central scale   *
+# For PDF uncertainty: use LHAPDF with supported set                   *
+#***********************************************************************
+ .true.   = reweight_scale   ! reweight to get scale dependence
+  0.5     = rw_Rscale_down   ! lower bound for ren scale variations
+  2.0     = rw_Rscale_up     ! upper bound for ren scale variations
+  0.5     = rw_Fscale_down   ! lower bound for fact scale variations
+  2.0     = rw_Fscale_up     ! upper bound for fact scale variations
+#True,True,False,False,False,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,False,False,True,False = reweight_PDF     ! reweight to get PDF uncertainty
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat:96]: please remove line 96

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define lep = e+ mu+ ta+ e- mu- ta-
+define nu = ve vm vt ve~ vm~ vt~
+define w = w+ w-
+generate p p > lep nu w a [QCD] @0
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_proc_card.dat:40]: you should also have a madspin card to decay the w

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+ 1.0  = xn        ! n parameter of eq.(3.4) in hep-ph/9801442
+ 1.0  = epsgamma  ! epsilon_gamma parameter of eq.(3.4) in hep-ph/9801442
+ .true.  = isoEM  ! isolate photons from EM energy (photons and leptons)
+#***********************************************************************
+# Maximal PDG code for quark to be considered a jet when applying cuts.*
+# At least all massless quarks of the model should be included here.   *
+#***********************************************************************
+ 4 = maxjetflavor
+#***********************************************************************
+4 = maxjetflavor
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat:148]: these two lines could be removed

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  2.6  = etal    ! Max lepton abs(pseudo-rap) (a value .lt.0 means no cut)
+   0  = drll    ! Min distance between opposite sign lepton pairs
+   0  = drll_sf ! Min distance between opp. sign same-flavor lepton pairs
+   0  = mll     ! Min inv. mass of all opposite sign lepton pairs
+  30  = mll_sf  ! Min inv. mass of all opp. sign same-flavor lepton pairs
+#***********************************************************************
+# Photon-isolation cuts, according to hep-ph/9801442                   *
+# When ptgmin=0, all the other parameters are ignored                  *
+#***********************************************************************
+  20  = ptgmin    ! Min photon transverse momentum
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat:136]: what's the ptg you plan to use in analysis, will 20GeV in GEN level be too high

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+#***********************************************************************
+   1  = jetalgo   ! FastJet jet algorithm (1=kT, 0=C/A, -1=anti-kT)
+ 0.7  = jetradius ! The radius parameter for the jet algorithm
+  15  = ptj       ! Min jet transverse momentum
+  -1  = etaj      ! Max jet abs(pseudo-rap) (a value .lt.0 means no cut)
+#***********************************************************************
+# Cuts on the charged leptons (e+, e-, mu+, mu-, tau+ and tau-)        *
+# (more specific gen cuts can be specified in SubProcesses/cuts.f)     *
+#***********************************************************************
+  20  = ptl     ! Min lepton transverse momentum
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat:126]: will 20 be too high in GEN level

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  2.6  = etal    ! Max lepton abs(pseudo-rap) (a value .lt.0 means no cut)
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/WWG/WWGToLNu2jG_4f_NLO_run_card.dat:127]: considering that the eta selection could be 2.5 for ele, use 3.0 could be safer

**Comment by attikis** [comment]
hi @menglu21 , @sv3048 , all,

do you mean that the GEN level pt threshold should be perhaps lowered? I would also agree with that suggestion. Besides, if thresholds are found to be too low for potential analyses aiming the high-pT phase space we can always extend the sample or even submit a complementary one that covers the desired phase space (w/ filters?).
Nevertheless, it is imperative that we submit these samples as soon as possible. 

Thank you,
Alexandros & Ho-Fung for HIG MC

**Comment by menglu21** [comment]
it's depending on the analysis requirement, we should always have looser cut in generator level than analysis level (reco-level). You can ask the analyser about it, if they are fine then we are ok too.

**Comment by attikis** [comment]
hi @menglu21 , 

we are in agreement in terms of GEN cut being looser than RECO-level requirement. 
If a particular analysis is not suited by this then we might explore some additional options, as stated above.
If @sv3048 and analysis teams also agree with this then we may proceed. 

Thank you,
Alexandros & Ho-Fung for HIG MC

**Comment by menglu21** [comment]
Hi, please address the comment to remove some unnecessary information.

**Pull request closed without merging**